docs/examples/MondrianTreeRegressor.md

```diff
@@ -0,0 +1,302 @@
+# Intuition behind Mondrian Trees
+
+This example provides intuition behind Mondrian Trees. In particular, the differences between existing decision tree algorithms and explanations of the tree construction and prediction will be highlighted.
+
+```python
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.tree import ExtraTreeRegressor
+from skgarden import MondrianTreeRegressor
+from itertools import cycle
+%matplotlib inline
+```
+
+##  A decision tree regressor
+
+A decision tree is one of the easier-to-understand machine learning algorithms. While training, the input training space `X` is recursively partitioned into a number of rectangular subspaces. While predicting the label of a new point, one determines the rectangular subspace that it falls into and outputs the label representative of that subspace. This is usually the mean of the labels for a regression problem.
+
+The rectangular subspaces are constructed in a greedy manner doing a binary partition at a time, hence each split is determined by a (f, \(\delta)\) tuple. The point `X` for which `X[f]` is lesser than \(\delta\) are in the left subspace and vice versa.
+
+## So how is each split done?
+
+The split tuple (f, \(\delta)\) is determined as that combination that creates child nodes such that the weighted decrease in impurity is maximum. Mathematically, this is the combination that maximises, `N_{parent} Imp_{Parent} - N_{left} Imp}_{left} - N_{right} Imp{right}`
+
+In a standard decision tree, this combination is found out by searching though all possible combinations of feature indices and values in the training data. That sounds pretty expensive!
```

Comment:
Importantly, this search is guided by a maximization of impurity decrease, which takes into account how better we would become for predicting the output.

I think this is important to underline, as mondrian trees skip this step.